server/rest/src/main/java/org/infinispan/rest/ALPNHandler.java

```diff
@@ -76,22 +78,56 @@ public void configurePipeline(ChannelPipeline pipeline, String protocol) {
     */
    protected void configureHttp2(ChannelPipeline pipeline) {
       pipeline.addLast(getHttp11To2ConnectionHandler());
+      addHttp1Handlers(pipeline);
+   }
+
+   /**
+    * Adds the commons Http1 handlers to the pipeline.
+    */
+   private void addHttp1Handlers(ChannelPipeline pipeline) {
+      // Handles http content encoding (gzip)
+      pipeline.addLast(new HttpContentCompressor(restServer.getConfiguration().getCompressionLevel()));
+      // Handles chunked data
+      pipeline.addLast(new HttpObjectAggregator(maxContentLength()));
+      // Handles Http/2 headers propagation from request to response
+      pipeline.addLast(new HeaderCorrelatorHandler());
+      // Handles CORS
+      pipeline.addLast(new CorsHandler(getCorsConfigs(), true));
+      // Handles Keep-alive
+      pipeline.addLast(new HttpServerKeepAliveHandler());
+      // Handles the writing of ChunkedInputs
       pipeline.addLast(new ChunkedWriteHandler());
-      pipeline.addLast("rest-handler-http2", new RestRequestHandler(restServer));
+      // Handles REST request
+      pipeline.addLast(new RestRequestHandler(restServer));
```

Comment:
is it required in this method? `configureHttp1()` adds another `RestRequestHandler` after this one.

Reply:
let me check

Reply:
I got rid of ```Http11RequestHandler``` altogether. It seems overkill to have an extra class just to handle an extra if